singularity-eos/eos/eos_spiner_sie_transforms.hpp

```diff
@@ -0,0 +1,214 @@
+#ifndef _SINGULARITY_EOS_EOS_EOS_SPINER_SIE_TRANSFORMS_HPP_
+#define _SINGULARITY_EOS_EOS_EOS_SPINER_SIE_TRANSFORMS_HPP_
+
+#ifdef SINGULARITY_USE_SPINER_WITH_HDF5
+
+#include <type_traits>
+
+#include <algorithm>
+#include <cstdlib>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <vector>
+// #include <iostream> // debug
+// #include <stdio.h> // debug
+
+// ports-of-call
+#include <ports-of-call/portability.hpp>
+
+// base
+#include <singularity-eos/eos/eos_spiner_common.hpp>
+
+namespace singularity {
+namespace transformations {
+
+
+
+template <typename Data = void>
+struct NullTransform {
+
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION NullTransform(Args &&...) {}
+
+  PORTABLE_INLINE_FUNCTION
+  NullTransform() = default;
+
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION auto transform(const Real e, Args &&...) const {
+    return e;
+  }
+
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION auto inverse(const Real e_transformed, Args &&...) const {
+    return e_transformed;
+  }
+};
+
+template <typename Data>
+struct ShiftTransform {
+  PORTABLE_INLINE_FUNCTION  ShiftTransform() = default;
+  
+  template <typename DataT_in>
+  PORTABLE_INLINE_FUNCTION ShiftTransform(const DataT_in &data) : data_(data) {}
+
+ private:
+  Data data_;
+
+ public:
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION auto transform(const Real e, const Real rho,
+                                          Args &&...) const {
+    const Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+    const Real e_cold = data_.sieCold.interpToReal(lRho);
+    return e - e_cold;
+  }
+
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION auto inverse(const Real e_transformed, const Real rho,
+                                        Args &&...) const {
+    const Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);   
+    const Real e_cold = data_.sieCold.interpToReal(lRho);
+    return e_transformed + e_cold;
+  }
+};
+
+
+
+//Divide by the heat capacity
+template<typename Data>
+struct DivideByCvTransform {
+  PORTABLE_INLINE_FUNCTION  DivideByCvTransform() = default;
+  
+  template <typename DataT_in>  
+  PORTABLE_INLINE_FUNCTION DivideByCvTransform(const DataT_in &data) : data_(data) {}
+
+  private: 
+    Data data_;
+    Real min_Cv_ = 1.0e-08;
```

Comment:
Make this compile-time. Also for reasons we'll discuss below, I think you want the inverse too.
```suggestion
   constexpr Real min_Cv_ = 1.0e-08;
   constexpr Real max_iCv_ = 1./1.e-8;
```